### ISSUE TITLE
Bug in the definition of the Tanimoto loss

### ISSUE THREAD

**Issue by feevos** [mohuazheliu/ResUnet-a#3]
Hi, thank you for developing code for the resunet-a model in Tensorflow. I am the leading author of the official ResUNet-a paper. 

There is a/some bug(s) in [your definition](https://github.com/ChienWong/ResUnet-a/blob/1d60d06d10f5cf3c5b15604b34963f0a8d4c5e4b/model.py#L19) of the Tanimoto loss. In particular, you need to sum along the channels and spatial dimensions, and also sum the predictions squared with the labels squared.  I assume that the predictions and labels are in dimensions (Batch, NClasses, Height, Width). 

I am not using actively TF, so some of the operations I will write here may not be valid, but they will make sense. 

This is correct (at least as pseudo-code): 
```python
# Here label.shape == Batch, NClasses, Height, Width, same for pred. 
def Tanimoto_loss(pred,label):
    # numerator, element wise multiplication 
    product=tf.multiply(pred,label).sum(axis=(1,2,3))

    # denominator 
    pred_square =tf.square(pred).sum(axis=(1,2,3))
    label_square = tf.square(label).sum(axis=(1,2,3))
    denum = pred_square + label_square - product 
  
    # for numerical stability (one way to achieve it)
    product = product + 1.e-4
    denum =  denum + 1.e-4
  
    tnmt = tf.divide(product, denum)
    loss = 1. - tnmt
    return loss # loss.shape == BatchSize
```
again, thank you for your time developing code.
